src/components/User/Operations/OperationDetails.vue

```diff
@@ -1,23 +1,72 @@
 <template>
-  <div class="operation-details">
-    <div class="app__block">
-      <h2>Operation details</h2>
-      <div class="operation-details__op-details-wrp">
+  <div v-if="operation.id">
+    <div class="operation-details">
+      <div class="app__block">
+        <h2>Operation details</h2>
         <details-reader :details="operation" />
       </div>
     </div>
   </div>
 </template>
 
 <script>
-import DetailsReader from '../../common/details/DetailsReader'
+import DetailsReader from '@comcom/details/DetailsReader'
+
+// ToDo: get operation by operationId from server
+import moment from 'moment'
+import get from 'lodash/get'
+import { ErrorHandler } from '@/utils/ErrorHandler'
+import { ApiCallerFactory } from '@/api-caller-factory'
+import { clearObject } from '@/utils/clearObject'
 
 export default {
   components: {
     DetailsReader,
   },
   props: {
-    operation: { type: Object, required: true },
+    operationId: { type: String, required: true },
+  },
+  data: _ => ({
+    userId: null,
+    operation: {},
+  }),
+  async created () {
+    this.userId = this.$route.params.id
+    await this.getOperations(this.operationId)
+  },
+  methods: {
+    async getOperations (operationId) {
+      let operationList = {}
+      try {
+        const response = await ApiCallerFactory
+          .createCallerInstance()
+          .getWithSignature('/v3/history', {
+            page: { order: 'desc' },
+            filter: clearObject({
+              account: this.userId,
+            }),
+            include: ['operation.details'],
+          })
+        operationList = response.data
+      } catch (error) {
+        ErrorHandler.processWithoutFeedback(error)
+      }
+      const operation = operationList
+        .find(operation => operation.id === this.operationId)
+      const operationType = operation.operation.details.type
+        .replace(/operations-/g, '')
+        .split('-')
+        .join(' ')
+      this.operation = Object.assign({}, operation, {
+        operationType: operationType
+          .charAt(0).toUpperCase() + operationType.slice(1),
```

Comment:
I think we can create some helper is `src/js/helpers` that will implement this feature. Also, it will more descriptive and reusable